azure/functions/decorators/utils.py

```diff
@@ -0,0 +1,125 @@
+        value fields. It is needed for enabling binding param optionality.
+        """
+        cls = super().__new__(mcs, name, bases, dct)
+        setattr(cls, '__init__',
+                cls.add_to_dict(getattr(cls, '__init__')))
+        setattr(cls, 'get_dict_repr',
+                cls.skip_none(getattr(cls, 'get_dict_repr')))
+        return cls
+
+    @staticmethod
+    def skip_none(func):
+        def wrapper(*args, **kw):
+            res = func(*args, **kw)
+            return BuildDictMeta.clean_nones(res)
+
+        return wrapper
+
+    @staticmethod
+    def add_to_dict(func: Callable):
+        def wrapper(*args, **kw):
+            if args is None or len(args) == 0:
+                raise ValueError(
+                    f'{func.__name__} has no args. Please ensure func is an '
+                    f'object method.')
+
+            func(*args, **kw)
+
+            setattr(args[0], 'init_params',
+                    list(inspect.signature(func).parameters.keys()))
+
+        return wrapper
+
+    @staticmethod
+    def clean_nones(value):
+        """
+        Recursively remove all None values from dictionaries and lists,
+        and returns
+        the result as a new dictionary or list.
+        """
+        if isinstance(value, list):
+            return [BuildDictMeta.clean_nones(x) for x in value if
+                    x is not None]
+        elif isinstance(value, dict):
+            return {
+                key: BuildDictMeta.clean_nones(val)
+                for key, val in value.items()
+                if val is not None
+            }
+        else:
+            return value
+
+
+class ABCBuildDictMeta(ABCMeta, BuildDictMeta):
+    pass
+
+
+def parse_singular_param_to_enum(param: Optional[Union[T, str]],
+                                 class_name: Type[T]) -> Optional[T]:
+    if param is None:
+        return None
+    if isinstance(param, str):
+        try:
+            return class_name[param]
+        except KeyError:
+            raise KeyError(
+                f"Can not parse str '{param}' to {class_name.__name__}. "
+                f"Allowed values are {[e.name for e in class_name]}")
+
+    return param
+
+
+def parse_iterable_param_to_enums(
+        param_values: Optional[Union[Iterable[str], Iterable[T]]],
+        class_name: Type[T]) -> Optional[Iterable[T]]:
+    if param_values is None:
+        return None
+
+    try:
+        return [class_name[value] if isinstance(value, str) else value for
+                value in param_values]
+    except KeyError:
+        raise KeyError(
+            f"Can not parse '{param_values}' to Optional[Iterable["
+            f"{class_name.__name__}]]. "
+            f"Please ensure param all list elements exist in "
+            f"{[e.name for e in class_name]}")
+
+
+def camel_case(snake_case: str):
+    words = snake_case.split('_')
+    return words[0] + ''.join(ele.title() for ele in words[1:])
+
+
+class CustomJsonEncoder(JSONEncoder):
```

Comment:
> Just a one-line comment - of why you needed a CustomJSONEncoder. Also, the update class name to `CustomJSONEncoder` to denote the subclass of the JSONEncoder.

Making enum str or int as first subclass i tried will make it a str or int type instead of enum, and 
```def parse_singular_param_to_enum(param: Optional[Union[T, str]], class_name: Type[T]) -> Optional[T]:``` would fail as no way to tell whether its a "GET" or HttpMethod.GET, they have same type str.